libs/fs.js

```diff
@@ -521,13 +521,15 @@ var fs = (function() {
       from = buffer.contentSize;
     }
 
-    var newLength = (from + data.byteLength > buffer.contentSize) ? (from + data.byteLength) : (buffer.contentSize);
+    var newLength = (from + length > buffer.contentSize) ? (from + length) : (buffer.contentSize);
 
     buffer.setSize(newLength);
 
-    buffer.array.set(data, from);
+    for (var i = 0; i < length; i++) {
```

Comment:
https://github.com/mozilla/j2me.js/issues/1356 is an issue I filed a while ago to implement this with a threshold.